### ISSUE TITLE
ordering problem: (migrate.js) Promise.all

### ISSUE THREAD

**Issue by Dmitry-N-Medvedev** [ThomWright/postgres-migrations#8]
The following line causes problems as far as I understand Promises

```
const completedMigrations = await Promise.all(
  filteredMigrations.map(
    runMigration(migrationTableName, client, log)
  )
)
```

... when:
* in 0001-init.sql I do `CREATE ROLE ... <role>`, and
* in 0002-create-db.sql I do `CREATE DATABASE ... WITH OWNER = <role>` 

Since `Promise.all` executes its statements in "parallel", the `OWNER` of the database created is always `postgres` probably because the `CREATE ROLE` transaction does not complete before the `CREATE DATABASE`.

I got success substituting the `completedMigrations = await Promise.all` with this:
```
const completedMigrations = [];

for (const migration of filteredMigrations) {
  const result = await runMigration(migrationTableName, client, log)(migration);
  completedMigrations.push(result);
}
```

**Comment by Dmitry-N-Medvedev** [comment]
remote: Permission to ThomWright/postgres-migrations.git denied to Dmitry-N-Medvedev.

**Comment by Dmitry-N-Medvedev** [comment]
#11

**Comment by ThomWright** [comment]
Fixed by #11
